artemis-cli/src/test/java/org/apache/activemq/artemis/cli/commands/CreateTest.java

```
@@ -0,0 +1,119 @@
+
+   @Parameters
+   public static Collection<Object[]> testData() {
+      return Arrays.asList(new Object[][]{
+         {"Happy path + relaxJolokia", "sampledomain.com", true},
+         {"Happy path - relaxJolokia", "sampledomain.net", false},
+         {"Domain with dash + relaxJolokia", "sample-domain.co", true},
+         {"Domain with dash - relaxJolokia", "sample-domain.co.uk", false},
+         {"Domain with double dashes + relaxJolokia", "sample--domain.name", true},
+         {"Domain with double dashes - relaxJolokia", "sample--domain.biz", false},
+         {"Domain with leading dashes + relaxJolokia", "--sampledomain.company", true},
+         {"Domain with leading dashes - relaxJolokia", "--sampledomain.email", false},
+         {"Domain with trailing dashes + relaxJolokia", "sampledomain--.shop", true},
+         {"Domain with trailing dashes - relaxJolokia", "sampledomain--.java", false},
+      });
+   }
+
+   @Test
+   public void testWriteJolokiaAccessXmlCreatesValidXml() throws Exception {
+      Create c = new Create();
+      String source = Create.ETC_JOLOKIA_ACCESS_XML;
+      HashMap<String, String> filters = new LinkedHashMap<>();
+
+      filters.put("${http.host}", this.httpHost);
+
+      // This is duplicated from Create.java, but it's embedded into the middle of the class.
+      // TODO: Refactor that code to it's own method.
+      if (this.relaxJolokia) {
+         filters.put("${jolokia.options}", "<!-- option relax-jolokia used, so strict-checking will be removed here -->");
+      } else {
+         filters.put("${jolokia.options}", "<!-- Check for the proper origin on the server side, too -->\n" + "        <strict-checking/>");
+      }
+
+      Path temp = Files.createTempFile("jolokia-access-", ".xml");
+      try {
+         c.write("etc/" + source, temp.toFile(), filters, false, true);
+
+         String xml = Files.readString(temp);
+         Assert.assertTrue(testName + " - should be valid, but isn't", isXmlValid(xml));
```

The xml check should not hide the failure cause
```suggestion
      try {
         XmlProvider.newDocumentBuilder().parse(new File(testInstance, "etc/" + Create.ETC_JOLOKIA_ACCESS_XML));
      } catch (Exception e) {
         Assert.fail(testName + " should be valid, but " + e.toString());
      }
```